src/components/common/Tooltip/Tooltip.tsx

```diff
@@ -1,29 +1,34 @@
+import { useState } from 'react';
 import styles from './tooltip.module.scss';
 
 type TooltipProps = {
-    isVisible: boolean;
-    textToShow: string;
-    tooltipPosition: {
-        top?: string;
-        right?: string;
-        bottom?: string;
-        left?: string;
-    };
+    children: React.ReactNode;
+    content: React.ReactNode;
+    tooltipPosition?: React.CSSProperties;
 };
 
-export default function Tooltip({
-    isVisible,
-    textToShow,
-    tooltipPosition,
-}: TooltipProps) {
+const Tooltip = ({ children, content, tooltipPosition }: TooltipProps) => {
+    const [currClass, setCurrClass] = useState('fade-out');
+    const toggleClass = () => {
+        if (currClass === 'fade-out') {
+            setCurrClass('fade-in');
+        } else {
+            setCurrClass('fade-out');
+        }
+    };
+
     return (
-        <span
-            data-testid="tooltip"
-            className={`${styles['tooltip']}
-               ${isVisible ? styles['fade-in'] : styles['fade-out']}`}
-            style={{ ...tooltipPosition }}
-        >
-            {textToShow}
+        <span onMouseOver={toggleClass} onMouseOut={toggleClass}>
+            {children}
+            <span
+                data-testid="tooltip"
+                className={`${styles['tooltip']} ${styles[currClass]} `}
+                style={{ ...tooltipPosition }}
+            >
+                {content}
+            </span>
```

Comment:
about about children will it be always visible since styles are only added to this span element

Reply:
No, it will not be always visible

Reply:
It will be visible please check once, resolving this because it's intended to be like this